bot/exts/evergreen/githubinfo.py

```diff
@@ -0,0 +1,97 @@
+import logging
+import random
+from datetime import datetime
+from typing import Optional
+
+import aiohttp
+import discord
+from discord.ext import commands
+from discord.ext.commands.cooldowns import BucketType
+
+from bot.constants import NEGATIVE_REPLIES
+
+log = logging.getLogger(__name__)
+
+
+class GithubInfo(commands.Cog):
+    """Fetches info from GitHub."""
+
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+
+    @staticmethod
+    async def fetch_data(url: str) -> dict:
+        """Retrieve data as a dictionary."""
+        async with aiohttp.ClientSession() as cs:
+            async with cs.get(url) as r:
+                return await r.json()
+
+    @commands.command(name='github', aliases=['gh'])
+    @commands.cooldown(1, 5, BucketType.user)
+    async def get_github_info(self, ctx: commands.Context, username: Optional[str]) -> None:
+        """
+        Fetches a user's GitHub information.
+
+        Username is optional and sends the help command if not specified.
+
+        Usage: .gh [username]
+        """
+        if username is None:
+            await ctx.invoke(self.bot.get_command('help'), 'github')
+            return
+
+        async with ctx.typing():
+            user_data = await self.fetch_data(f"https://api.github.com/users/{username}")
+
+            # User_data will not have a message key if the user exists
+            if user_data.get('message') is not None:
+                await ctx.send(embed=discord.Embed(title=random.choice(NEGATIVE_REPLIES),
+                                                   colour=discord.Colour.red()))
+                return
+
+            org_data = await self.fetch_data(user_data['organizations_url'])
+            orgs = [f"[{org['login']}](https://github.com/{org['login']})" for org in org_data]
+            orgs_to_add = ' | '.join(orgs)
+
+            starred_data = await self.fetch_data(user_data['starred_url'])
```

Comment:
This request returns a 404 with the following response:
```json
{"message": "Not Found", "documentation_url": "https://docs.github.com/rest"}
```
This is because `user_data['starred_url']` returns this link: `https://api.github.com/users/USER/starred{/owner}{/repo}`. The `{/owner}{/repo}` are optional and should be replaced with actual values if needed. In this case, we don't need them. 

I'd recommend to either do this:
```suggestion
            # Removed the optional `{/owner}{/repo}`
            starred_data = await self.fetch_data(user_data['starred_url'][:-15])
```
Or this:
```suggestion
            starred_data = await self.fetch_data(f"https://api.github.com/users/{username}/starred")
```

<hr>

Another issue here is that this only returns a maximum of 30 starred repositories. We can increase that to 100 per request, but this won't be accurate if the user has over 100 starred repositories. Do we think that's okay? Or should we remove this statistic and possibly replace it with a different one? The amount of public gists could work.

Reply:
Hmmm, I didn't know about the 30 starred repo limit thing, thanks for catching that. Instead of displaying inaccurate stats I think we should just replace it with gists, that sounds like a great idea to me

Just out of curiosity, how would I increase that to 100 per request?

Reply:
You can add on `?per_page=100` to the end of the URL to get 100 per request. Some endpoints default to 30, and others default to 100. The starred repositories endpoint happens to default to 30.